src/http/adapter.js

```diff
@@ -68,6 +68,14 @@ function checkPermissionFor( state, user, context, action ) {
 		.then( onPermission, onError );
 }
 
+function executeStack( resource, envelope, stack ) {
+	var calls = stack.slice( 0 );
+	var call = calls.shift();
+	return call ?
+		call.bind( resource )( envelope, executeStack.bind( undefined, resource, envelope, calls ) ) :
+		{ status: 500, data: { message: 'The server failed to provide a response' } };
```

Comment:
That would mean that middleware is required on all resources and actions, don't want that. The catch is almost silly to have, because you'd have to specifically create a middleware stack that included undefined as an array element. It's mostly there in the event that you are referencing a function by an invalid name. I can see throwing/logging an error I suppose but I didn't want it to result in a timeout for the client as an error reported to us as a timeout doesn't indicate the issue quite as clearly as this error message that only occurs should we provide an undefined function in a stack.